flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBNativeMetricMonitor.java

```diff
@@ -231,4 +252,88 @@ public void update() {
             setStatistics(this);
         }
     }
+
+    /**
+     * A gauge which periodically pulls a RocksDB statistics-based native statistic metric for the database.
+     */
+    class RocksDBNativeHistogramStatisticsMetricView extends RocksDBNativeView implements Histogram {
+        private final HistogramType histogramType;
+
+        private HistogramData histogramData;
+
+        private RocksDBNativeHistogramStatisticsMetricView(HistogramType histogramType) {
+            this.histogramType = histogramType;
+        }
+
+        void setValue(HistogramData histogramData) {
+            this.histogramData = histogramData;
+        }
+
+        @Override
+        public void update() {
+            setHistogramStatistics(this);
+        }
+
+        @Override
+        public void update(long value) {
+        }
+
+        @Override
+        public long getCount() {
+            return histogramData == null ? 0 : histogramData.getCount();
+        }
+
+        @Override
+        public HistogramStatistics getStatistics() {
+            return new RocksDBNativeHistogram();
+        }
+
+        class RocksDBNativeHistogram extends HistogramStatistics {
+
+            @Override
+            public double getQuantile(double quantile) {
```

Comment:
Flink would have `p50`, `p75`, `p95`, `p98`, `p99` and `p999` quantiles, I don't think returning zero would be a good idea here.